network/alsp/manager/manager.go

```diff
@@ -428,3 +447,65 @@ func (m *MisbehaviorReportManager) processMisbehaviorReport(report internal.Repo
 	lg.Debug().Float64("updated_penalty", updatedPenalty).Msg("misbehavior report handled")
 	return nil
 }
+
+// adjustDecayFunc calculates the decay value of the spam record cache. This allows the decay to be different on subsequent disallow listings.
+// It returns the decay speed for the given cutoff counter.
+// The cutoff counter is the number of times that the node has been disallow-listed.
+// Args:
+// cutoffCounter: the number of times that the node has been disallow-listed including the current time. Note that the cutoff counter
+// must always be updated before calling this function.
+//
+// Returns:
+//
+//	float64: the decay speed for the given cutoff counter.
+func (m *MisbehaviorReportManager) adjustDecayFunc(cutoffCounter uint64) float64 {
+	// decaySpeeds illustrates the decay speeds for different cutoff counters.
+	// The first cutoff does not reduce the decay speed (1000 -> 1000). However, the second, third,
+	// and forth cutoffs reduce the decay speed by 90% (1000 -> 100, 100 -> 10, 10 -> 1).
+	// All subsequent cutoffs after the fourth cutoff use the last decay speed (1).
+	// This is to prevent the decay speed from becoming too small and the spam record from taking too long to decay.
+	decaySpeeds := []float64{1000, 1000, 100, 10, 1}
+
+	if cutoffCounter <= 0 {
+		// illegal state, this should never happen unless there is a bug in the code.
+		panic(fmt.Sprintf("illegal-state cutoff counter must be positive, it should include the current time: %d", cutoffCounter))
+	}
+
+	if int(cutoffCounter) >= len(decaySpeeds) {
+		return decaySpeeds[len(decaySpeeds)-1] // clamp to the last value
+	}
+
+	return decaySpeeds[cutoffCounter]
```

Comment:
nit: Since we are hard coding the decay speeds already and pretty much hard coding the cutoff counter intervals (1, 2, 3, 4) we can simplify this code block. Just a suggestion I leave it up to your judgement. 
```suggestion
	switch {
	case cutoffCounter == 1:
		return 1000
	case cutoffCounter == 2:
		return 100
	case cutoffCounter == 3:
		return 10
	case cutoffCounter >= 4:
		return 1
	default:
		panic(fmt.Sprintf("illegal-state cutoff counter must be positive, it should include the current time: %d", cutoffCounter))
	}
```